### PR TITLE
Add Calculation block package

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

Adding the F# calculation block to the BP hub

Semi-blocked on https://github.com/hashintel/hash/pull/683 to bump block-scripts here as well

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202250835094705/1202427004949349/f) _(internal)_

## 🔍 What does this change?

- Adds a 'calculation' folder to the blocks package containing a thin wrapper around the block
- The block source is still at https://github.com/hashdeps/calculation-table-block

## 🐾 Next steps

- Change the styling of the block to match the rest of the blocks. Difficult as the MUI v5 isn't implemented in Fable (yet)
- Better README, as it will be shown on the hub at some point.

Some hiccups were found while adding the block, which will be addressed in follow-up PRs:
- JS Bundle size is unecesarrily big because of [inlined source maps](https://github.com/blockprotocol/blockprotocol/blob/bbf01fe1ea9241e6bdf2cd6490981464de6ba560/packages/block-scripts/shared/webpack-config.js#L110)
- Default react block-app template [entrypoint value](https://github.com/blockprotocol/blockprotocol/blob/bbf01fe1ea9241e6bdf2cd6490981464de6ba560/packages/block-template/templates/react/package.json#L44) doesn't match up with what's expected in the [Block Hub](https://github.com/blockprotocol/blockprotocol/blob/bbf01fe1ea9241e6bdf2cd6490981464de6ba560/site/src/pages/api/rewrites/sandboxed-block-demo.api.ts#L146)

## ❓ How to test this?

1. Checkout branch
2. in the new directory, run `yarn && yarn dev`

## 📹 Demo

image

### CODE REVIEW

File: packages/blocks/calculation/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Calculation
+
```
Comment (kachkaev) [inline packages/blocks/calculation/README.md:2]: Let’s remove this H1 as in other blocks (this improves README display in block hub)

File: packages/blocks/calculation/package.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/hashintel/hash.git",
+    "directory": "packages/blocks/calculation"
+  },
+  "license": "MIT",
+  "author": "Ahmad Sattar Atta",
+  "scripts": {
+    "build": "node_modules/.bin/block-scripts build",
+    "dev": "node_modules/.bin/block-scripts dev",
```
Comment (kachkaev) [inline packages/blocks/calculation/package.json:18]:
```suggestion
    "build": "block-scripts build",
    "dev": "block-scripts dev",
```

File: packages/blocks/calculation/package.json [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    "fix:eslint": "eslint --ext .ts,.tsx --fix ./src/",
+    "lint:eslint": "eslint --ext .ts,.tsx ./src/",
+    "lint:tsc": "tsc --noEmit",
+    "serve": "node_modules/.bin/block-scripts serve"
```
Comment (kachkaev) [inline packages/blocks/calculation/package.json:22]:
```suggestion
    "build": "block-scripts build",
    "dev": "block-scripts dev",
    "fix:eslint": "eslint --ext .ts,.tsx --fix ./src/",
    "lint:eslint": "eslint --ext .ts,.tsx ./src/",
    "lint:tsc": "tsc --noEmit",
    "serve": "block-scripts serve"
```

Comment (thehabbos007) [reply packages/blocks/calculation/package.json:22]: Oh, i see the other blocks do the same. Is this specific to our monorepo setup, or could this be changed in the react template as well?

Comment (kachkaev) [reply packages/blocks/calculation/package.json:22]: Not sure why we'd need`node_modules/.bin` prefix in both cases. Mentioning a dependency via `yarn xyz` should resolve the binary. There is no guarantee that the file will be in `node_modules/.bin` in a general case scenario.

File: packages/blocks/calculation/src/types.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+declare module "calculation-block";
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/blocks/calculation/src/types.d.ts:1]: Yeah, it's just basic JS at this point. We should definitely get better typings from the NPM package itself

[the remark this answers is not in the document]
Comment (kachkaev) [reply packages/blocks/calculation/src/types.d.ts:1]: We can still provide typings from the package if we define them manually. Strictly speaking, there is no need to transpile ts files to js 🙂

**Pull request merged**